### PR TITLE
add the instruction to install cluster local gateway

### PR DESCRIPTION
**Pull request by ZhiminXiang**
## Proposed Changes

add the instruction to install cluster local gateway

This PR just cherrypicks the change https://github.com/knative/docs/pull/1894 into master branch

### CODE REVIEW

**Comment by ZhiminXiang** [comment]
friendly ping

**Comment by RichieEscarez** [comment]
@ZhiminXiang ill try to get to this later today, thanks!

File: docs/install/installing-istio.md [hunk 1]
```diff
@@ -216,33 +216,51 @@ helm template --namespace=istio-system \
...
-  --set gateways.istio-egressgateway.enabled=false \
-  --set gateways.istio-ilbgateway.enabled=false \
-  install/kubernetes/helm/istio \
-  -f install/kubernetes/helm/istio/example-values/values-istio-gateways.yaml \
-  | sed -e "s/custom-gateway/cluster-local-gateway/g" -e "s/customgateway/clusterlocalgateway/g" \
-  > ./istio-local-gateway.yaml
-
-kubectl apply -f istio-local-gateway.yaml
-```
+### Installing `cluster-local-gateway` for internal cluster-only routes
```
Comment (RichieEscarez) [inline docs/install/installing-istio.md]: I editing this section for clarity and organization. PTAL and ensure its accuracy.

File: docs/install/Knative-custom-install.md [hunk 2]
```diff
@@ -49,6 +49,10 @@ Minkube or similar, see the
 > [Click here](./Knative-with-Ambassador.md) to install Knative with Ambassador.
 > [Click here](./Knative-with-Gloo.md) to install Knative with Gloo.
 
+## Installing Cluster Local Gateway
```
[the remark this answers is not in the document]
Comment (ZhiminXiang) [reply docs/install/Knative-custom-install.md]: > Global: The way these statements are added here across the install guides it reads like this is a required step? Or is this optional?
In general, `cluster-local-gateway` is not required only when users do not need to send and serve cluster-internal traffic.
Specifically, for internal cluster-only routes, this is required as all traffic is cluster internal.
For public routes, if users do not need to send cluster-internal traffic, it is not required. But I think we may want to recommend them to install it in order to provide them consistent user experience.

So my recommendation is rewording it to "Installing `cluster-local-gateway` for serving cluster-internal traffic", and explain that it is required for private routes and optional (but recommended) for public routes. WDYT?

> 
> Is this accurate? We, nor does Istio, define what a "Cluster Local Gateway" is so im trying to better explain what this user-focused task is here:

Yes `cluster-local-gateway` is defined by Knative, not Istio.

File: docs/install/Knative-custom-install.md [hunk 3]
```diff
@@ -49,6 +49,10 @@ Minkube or similar, see the
...
+
+Follow the [instruction](./installing-istio#updating-your-install-to-use-cluster-local-gateway) to install Cluster Local Gateway.
```
[the remark this answers is not in the document]
Comment (ZhiminXiang) [reply docs/install/Knative-custom-install.md]: I woud reword it to something like
```
If you installed Istio, you can install a cluster-local-gateway within Knative cluster in order to be able to serve cluster-internal traffic.
```
and then point to the instruction.

File: docs/install/installing-istio.md [hunk 4]
```diff
@@ -216,33 +216,51 @@ helm template --namespace=istio-system \
...
+
+You can configure your Knative cluster to use an 
+[Istio gateway](https://istio.io/docs/reference/config/networking/v1alpha3/gateway/) that routes 
+all traffic internally. To be able to configure your revisions to use routes that are visible only within your cluster, 
+you install and use the `cluster-local-gateway` gateway.
```
Comment (ZhiminXiang) [inline docs/install/installing-istio.md]: I left a comment above which also works for this part.

**Comment by samodell** [comment]
@ZhiminXiang  Let's make a few changes and then get this in:

Update the heading on all pages to:

`###Installing `cluster-local-gateway` for serving cluster-internal traffic`

And then change the text below to:

```
If you installed Istio, you can install a `cluster-local-gateway` within your Knative cluster so that you can serve cluster-internal traffic. If you want to configure your revisions to use routes that are visible only within your cluster, [install and use the `cluster-local-gateway`](./installing-istio#updating-your-install-to-use-cluster-local-gateway).
```

----------------

These changes combines Richie's suggestions with your suggestions. :)

File: docs/install/installing-istio.md [hunk 5]
```diff
@@ -244,6 +244,22 @@ helm template --namespace=istio-system \
 kubectl apply -f istio-local-gateway.yaml
 ```
 
+Alternatively, if you want to install cluster local gateway for **development purpose**, run below command
```
Comment (ZhiminXiang) [inline docs/install/installing-istio.md]: @samodell I accidentally lost Richie's  suggestion. Could you let me know how the change in this file looks?

Comment (samodell) [inline docs/install/installing-istio.md]:
```suggestion
Alternatively, if you want to install the cluster local gateway for **development purposes**, enter the following command 
```

File: docs/install/installing-istio.md [hunk 6]
```diff
@@ -244,6 +244,22 @@ helm template --namespace=istio-system \
...
+```shell
+# Istio minor version should be 1.2 or 1.3
+export ISTIO_MINOR_VERSION=1.2
+
+export VERSION=$(curl https://raw.githubusercontent.com/knative/serving/master/third_party/istio-${ISTIO_MINOR_VERSION}-latest)
+
+kubectl apply -f https://raw.githubusercontent.com/knative/serving/master/third_party/${VERSION}/istio-knative-extras.yaml
+```
+
+**Note:** Above installation method is only for development purpose. The production readiness of above
```
Comment (samodell) [inline docs/install/installing-istio.md]:
```suggestion
**Note:** This method is only for development purposes. The production readiness of the above 
```

File: docs/install/installing-istio.md [hunk 7]
```diff
@@ -244,6 +244,22 @@ helm template --namespace=istio-system \
...
+# Istio minor version should be 1.2 or 1.3
+export ISTIO_MINOR_VERSION=1.2
+
+export VERSION=$(curl https://raw.githubusercontent.com/knative/serving/master/third_party/istio-${ISTIO_MINOR_VERSION}-latest)
+
+kubectl apply -f https://raw.githubusercontent.com/knative/serving/master/third_party/${VERSION}/istio-knative-extras.yaml
+```
+
+**Note:** Above installation method is only for development purpose. The production readiness of above 
+installation method is not ensured. For production installation, refer the `helm` installation method in this
```
Comment (samodell) [inline docs/install/installing-istio.md]:
```suggestion
installation method is not ensured. For a production-ready installation, see the `helm` installation method above.
```

File: docs/install/installing-istio.md [hunk 8]
```diff
@@ -244,6 +244,22 @@ helm template --namespace=istio-system \
...
+section.
```
Comment (samodell) [inline docs/install/installing-istio.md]:
```suggestion
```
